### PR TITLE
Create ci.yml GitHub action to build, test

### PR DESCRIPTION
**Pull request by cruizen**
@Rohitrajak1807 This may look familiar. :-)

### CODE REVIEW

**Review by mithilarun — APPROVED** [review]

**Comment by Rohitrajak1807** [comment]
> @Rohitrajak1807 This may look familiar. :-)

It's the same from arlon, but we're not including the golang ci lint action. We could include it later when it's needed.

**Review by Rohitrajak1807 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Unit Tests and Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (Rohitrajak1807) [inline .github/workflows/ci.yml:9]: Do we not have a similar per-release branch here too?

**Review by Rohitrajak1807 — COMMENTED** [review]
Left a comment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cruizen) [reply .github/workflows/ci.yml:9]: I am not sure what other branches we should include (we definitely exclude the private/* branches but shall we include the platform9/* branches? There are only two). 
 
IMO, let's get started with master and then we can add branches later.

**Comment by cruizen** [comment]
> > @Rohitrajak1807 This may look familiar. :-)
> 
> It's the same from arlon, but we're not including the golang ci lint action. We could include it later when it's needed.

Yes, that part is commented since the codecov app needs permissions for the pf9 repos. It's a longer loop than Arlon to get this in place.

**Comment by Rohitrajak1807** [comment]
> > > @Rohitrajak1807 This may look familiar. :-)
> > 
> > 
> > It's the same from arlon, but we're not including the golang ci lint action. We could include it later when it's needed.
> 
> Yes, that part is commented since the codecov app needs permissions for the pf9 repos. It's a longer loop than Arlon to get this in place.

But I see it ran.

**Comment by cruizen** [comment]
> > > > @Rohitrajak1807 This may look familiar. :-)
> > > 
> > > 
> > > It's the same from arlon, but we're not including the golang ci lint action. We could include it later when it's needed.
> > 
> > 
> > Yes, that part is commented since the codecov app needs permissions for the pf9 repos. It's a longer loop than Arlon to get this in place.
> 
> But I see it ran. https://github.com/platform9/luigi/actions/runs/4178073771/jobs/7236436784

And then I commented it out for now since we need to fix the lint issues too. You are welcome to take that up as a follow-up.

**Pull request merged**